src/routes/profile-route/profile-settings-route.jsx

```diff
@@ -0,0 +1,242 @@
+import { useMutation, useQuery } from "@apollo/client";
+import { useContext, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useMediaQuery } from "react-responsive";
+
+//import { DeleteUserModal } from "../../components/delete-user-modal";
+import {
+  DeleteConfirmationModal,
+  Footer,
+  Form,
+  Header,
+} from "../../components/index";
+import { DELETE_USER, GET_USER, GET_USERS, UPDATE_USER } from "../../graphql/";
+import { useModals } from "../../store";
+import { CurrentUserContext } from "../../store/current-user-context";
+import { ProfileHorizontalMenu, ProfileSidebar } from "./index";
+
+export const ProfileSettingsRoute = () => {
+  const { currentUser, setCurrentUser } = useContext(CurrentUserContext);
+  const userId = currentUser.id;
+  const { registerModal, openModal, closeModal } = useModals();
+  const [deleteUser] = useMutation(DELETE_USER);
+  const prefilledInputs = {
+    username: currentUser?.username,
+    email: currentUser?.email,
+  };
+
+  const userQuery = useQuery(GET_USER, {
+    variables: {
+      where: {
+        id: userId,
+      },
+    },
+  });
+
+  const usersQuery = useQuery(GET_USERS, { variables: { where: {} } });
+
+  const [updateUser, { data, loading, error }] = useMutation(UPDATE_USER, {
+    refetchQueries: [GET_USER],
+  });
+
+  const handleDeleteUserSubmit = () => {
+    //loading visible
+    //call deleteUser(userID)
+    deleteUser({
+      variables: {
+        where: { id: userId },
+      },
+    })
+      //.then(res)
+      .then(({ data }) => {
+        //console.log(data);
+        // - - - delete stored token
+        localStorage.removeItem("jwt");
+        // - - - close modal
+        closeModal("deleteUser");
+        window.location.reload();
+        // - - - remove "loading"
+        // - - - redirect to home page
+        // - - - or run "logout" when that is setup
+      })
+      //.catch
+      .catch(error);
+    //.finally?
+  };
+
+  // - - - - - - - - - - USE EFFECT - - - - - - - - - -
+  useEffect(() => {
+    registerModal(
+      "deleteUser",
+      <DeleteConfirmationModal
+        onClose={closeModal}
+        confirmButtonAction={handleDeleteUserSubmit}
+      />
+    );
+  }, []);
+
+  // - - - - - - - - - - USE FORM - - - - - - - - - -
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    setValue,
+    getValues,
+  } = useForm({
+    defaultValues: {
+      username: "",
+      email: "",
+      password: "",
+    },
+  });
+
+  const onSubmit = () => {
```

Comment:
My username doesn't update when I click "Save". I haven't tried password yet. It appears to be giving no console log when hitting the submit button.

Reply:
This actually might be due to how I have my user set up? I tried to delete my user and it said I don't have access. So I may need someone else to test it out.

Reply:
@fostert14 Thanks for your input here! Right now the backend is setup to only allow admins to do certain updates. So in its current state you will only be able to test this out as an admin user. I also have PR waiting for review for the backend repository for this exact reason.